proposals/testing/NNNN-polling-confirmations.md

```diff
@@ -0,0 +1,512 @@
+  ///     running.
+  ///     if nil, polling will be attempted for approximately 1 second.
+  ///     `duration` must be greater than 0.
+  ///   - interval: The minimum amount of time to wait between polling
+  ///     attempts.
+  ///     If nil, polling will wait at least 1 millisecond between polling
+  ///     attempts.
+  ///     `interval` must be greater than 0.
+  public static func pollingUntilStopsPassingDefaults(
+    until duration: Duration? = nil,
+    pollingEvery interval: Duration? = nil
+  ) -> Self
+}
+```
+
+Specifying `duration` or `interval` directly on either new `confirmation`
+function will override any value provided by the relevant trait. Additionally,
+when multiple of these configuration traits are specified, the innermost or
+last trait will be applied.
+
+### Default Polling Configuration
+
+For all polling confirmations, the Testing library will default `duration` to
+1 second, and `interval` to 1 millisecond.
+
+### Platform Availability
+
+Polling confirmations will not be available on platforms that do not support
+Swift Concurrency.
+
+### Duration and Concurrent Execution
+
+It is an unfortunate side effect that directly using the `duration` to determine
+when to stop polling (i.e. `while duration has not elapsed { poll() }`) is
+unreliable in a parallel execution environment. Especially on systems that are
+under-resourced, under very high load, or both - such as CI systems. This is
+especially the case for the Testing library, which, at time of writing, submits
+every test at once to the concurrency system for scheduling. Under this
+environment, with heavily-burdened machines running test suites with a very
+large amount of tests, there is a very real case that a polling confirmation's
+`duration` might elapse before the `body` has had a chance to return even once.
+
+To prevent this, the Testing library will calculate how many times to poll the
+`body`. This is done by dividing the `duration` by the `interval`. For example,
+with the default 1 second duration and 1 millisecond interval, the Testing
+library will poll 1000 times, waiting 1 millisecond between polling attempts.
+This works and is immune to the issues posed by concurrent execution on
+heavily-burdened systems.
+This is also very easy for test authors to understand and predict, even if it is
+not fully accurate - each poll attempt takes some amount of time, even for very
+fast `body` closures. Which means that the real-time duration of a polling
+confirmation will always be longer than the value specified in the `duration`
+argument.
+
+### Usage
+
+These functions can be used with an async test function:
+
+```swift
+@Test func `The aquarium's dolphin nursery works`() async {
+    let subject = Aquarium()
+    Task {
+        await subject.raiseDolphins()
+    }
+    await confirmation(until: .firstPass) {
+        await subject.dolphins.count == 1
+    }
+}
+```
+
+With the definition of `Aquarium` above, the closure will only need to be
```

Comment:
nit: "may" only need? Since technically you can't predict or guarantee any particular timing; the system could be under heavy load
```suggestion
With the definition of `Aquarium` above, the closure may only need to be
```